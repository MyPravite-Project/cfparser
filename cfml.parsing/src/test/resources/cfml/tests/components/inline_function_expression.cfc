component {

	function dotrim(data, callback){
		callback(trim(data));
	}

	function bar(){
		dotrim(function("hello world"){
			//...
		});
	}

}
