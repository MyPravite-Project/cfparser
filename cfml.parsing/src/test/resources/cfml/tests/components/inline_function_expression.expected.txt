/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <dotrim>
'('                      <(>
IDENTIFIER               <data>
','                      <,>
IDENTIFIER               <callback>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
IDENTIFIER               <callback>
'('                      <(>
IDENTIFIER               <trim>
'('                      <(>
IDENTIFIER               <data>
')'                      <)>
')'                      <)>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <bar>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
IDENTIFIER               <dotrim>
'('                      <(>
OPEN_STRING              <">
STRING_LITERAL           <hello world>
CLOSE_STRING             <">
','                      <,>
FUNCTION                 <function>
'('                      <(>
IDENTIFIER               <result>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
'}'                      <}>
')'                      <)>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier dotrim)
          (
          (parameterList
            (parameter (identifier data))
            ,
            (parameter (identifier callback))
          )
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifier callback)
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (unaryExpression
                                      (memberExpression
                                        (functionCall
                                          (identifier trim)
                                          (
                                          (argumentList
                                            (argument
                                              (startExpression
                                                (compareExpression
                                                  (baseExpression (unaryExpression (memberExpression (identifier data))))
                                                )
                                              )
                                            )
                                          )
                                          )
                                        )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      (element
        (functionDeclaration
          function
          (identifier bar)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifier dotrim)
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (unaryExpression
                                      (primaryExpression
                                        (literalExpression (stringLiteral " (stringLiteralPart hello world) "))
                                      )
                                    )
                                  )
                                )
                              )
                            )
                            ,
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (anonymousFunctionDeclaration
                                      function
                                      (
                                      (parameterList (parameter (identifier result)))
                                      )
                                      (compoundStatement { })
                                    )
                                  )
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  {
    public function bar()       {
dotrim('hello world',  function (result)   {

  });

      };
    public function dotrim(data, callback)       {
callback(trim(data));

      };

  }
}
/*======*/